### PR TITLE
Fix pylint global-statement

### PR DESCRIPTION
**Pull request by marcospri**

### CODE REVIEW

File: h/accounts/schemas.py [hunk 1]
```diff
@@ -19,23 +20,20 @@
 log = logging.getLogger(__name__)
 
 PASSWORD_MIN_LENGTH = 2  # FIXME: this is ridiculous
-USERNAME_BLACKLIST = None
 
 
+@lru_cache(maxsize=None)
```
Comment (marcospri) [inline h/accounts/schemas.py:25]: According to the docs these should be marginally faster than maxsize=1 (removes the eviction logic)

File: h/auth/__init__.py [hunk 2]
```diff
@@ -36,6 +36,7 @@
 
 
 def includeme(config):
+    # pylint: disable=global-statement
     global DEFAULT_POLICY
     global WEBSOCKET_POLICY
```
Comment (jon-betts) [inline h/auth/__init__.py:41]: I'll be trying to fix this soon :crying_cat_face:

File: h/search/parser.py [hunk 3]
```diff
@@ -90,13 +90,7 @@ def unparse(q):
...
 
 
-def _get_parser():
-    global parser
-    if parser is None:
-        parser = _make_parser()
-    return parser
-
-
+@lru_cache(maxsize=None)
```
Comment (jon-betts) [inline h/search/parser.py:93]: :+1:

File: h/util/markdown.py [hunk 4]
```diff
@@ -135,7 +135,7 @@ def _linkify_rel(attrs, new=False):  # pylint: disable=unused-argument
 
 
 def _get_cleaner():
-    global cleaner
+    global cleaner  # pylint: disable=global-statement
```
Comment (jon-betts) [inline h/util/markdown.py:138]: I think the `lru_cache` solution above was much nicer and would work here too.

File: h/accounts/schemas.py [hunk 5]
```diff
@@ -19,23 +20,20 @@
...
+    # Try to load the blacklist file from disk. If, for whatever reason, we
+    # can't load the file, then don't crash out, just log a warning about
+    # the problem.
+    try:
+        with codecs.open("h/accounts/blacklist", encoding="utf-8") as fp:
+            blacklist = fp.readlines()
+    except (IOError, ValueError):
+        log.exception("unable to load blacklist")
+        blacklist = []
+    return set(line.strip().lower() for line in blacklist)
```
Comment (jon-betts) [inline h/accounts/schemas.py:36]: So the `USERNAME_BLACKLIST` is just a cache here? Noice

**Review by jon-betts — APPROVED** [review]
This is good, although I think I like the `lru_cache` solution to solve the last two as well as the first two.

**Pull request merged**